src/pages/Facility/queues/QueuesIndex.tsx

```diff
@@ -259,23 +258,40 @@ export default function QueuesIndex({
     }
   }, [qParams.date, updateQuery]);
 
-  // Handle date filter
-  const handleDateChange = (date: Date | undefined) => {
-    if (date) {
-      updateQuery({ date: dateQueryString(date) });
-    } else {
-      updateQuery({ date: undefined });
-    }
-  };
+  // Handle date filter - memoized to prevent unnecessary re-renders
+  const handleDateChange = useCallback(
+    (date: Date | undefined) => {
+      if (date) {
+        updateQuery({ date: dateQueryString(date) });
+      } else {
+        updateQuery({ date: undefined });
+      }
+    },
+    [updateQuery],
+  );
 
-  // Handle resource selection
-  const handleResourceChange = (selectedResourceId: string) => {
-    updateQuery({ resource_id: selectedResourceId });
-  };
+  // Handle resource selection - memoized to prevent unnecessary re-renders
+  const handleResourceChange = useCallback(
+    (users: UserReadMinimal[]) => {
+      if (users.length > 0) {
+        updateQuery({ resource_id: users[0].id });
+      } else {
+        // Clear the selection and reset to current user's ID
+        updateQuery({ resource_id: undefined });
+      }
```

Comment:
isn't this same as this? 🤔 

if 0th element doesn't exist, it'd be undefined right?

```suggestion
      updateQuery({ resource_id: users[0]?.id });
```